custom/icds_reports/utils/aggregation_helpers/distributed/agg_gov_dashboard.py

```diff
@@ -147,5 +153,66 @@ def update_queries(self):
         DROP TABLE temp_gov_dashboard;
         """, {}
 
+        yield """
+        CREATE TABLE temp_cbe_data AS
+            SELECT
+                ucr.awc_id,
+                ucr.state_id,
+                ucr.date_cbe_organise,
+                ucr.theme_cbe,
+                ucr.count_targeted_beneficiaries,
+                ucr.count_other_beneficiaries,
+                rank() OVER (
+                    PARTITION BY awc_id
+                    ORDER BY date_cbe_organise, submitted_on
+                    )
+
+            FROM "ucr_icds-cas_static-cbe_form_f7988a04" ucr
+            WHERE ucr.date_cbe_organise>=%(start_date)s AND ucr.date_cbe_organise<%(next_month_start_date)s;
+        """.format(
+            cbe_ucr_table=get_table_name(self.domain, 'static-cbe_form')
+        ), {
+            'start_date': self.month,
+            'next_month_start_date': next_month_start
+        }
+
+        yield """
+        UPDATE "{tmp_tablename}" gov_table
+        SET cbe_conducted_1= ut.cbe_conducted_1,
+            cbe_type_1=ut.cbe_type_1,
+            cbe_date_1=ut.cbe_date_1,
+            num_target_beneficiaries_1=ut.num_target_beneficiaries_1,
+            num_other_beneficiaries_1=ut.num_other_beneficiaries_1,
+            cbe_conducted_2=ut.cbe_conducted_2,
+            cbe_type_2=ut.cbe_type_2,
+            cbe_date_2=ut.cbe_date_2,
+            num_target_beneficiaries_2=ut.num_target_beneficiaries_2,
+            num_other_beneficiaries_2=ut.num_other_beneficiaries_2
+        FROM
+        (
+        SELECT
+            awc_id,
+            MIN(CASE WHEN rank=1 THEN 'yes' END) as cbe_conducted_1,
```

Comment:
This Min is not finding the min. Its just helping to group by awc_id to get 1 row by awc_id

Reply:
How did you choose `min`? Does there not need to be some kind of conflict resolution algorithm?

Reply:
I could have chose Max. That gives the same result. 
what kind of conflict resolution could you explain a bit more?

Reply:
Like if there are multiple values and one is "yes" and the other is "no" then max/min would result in a different agg value, no? Does that not matter?

Reply:
hey, that only affects my  result when I write the staetment like `MIN(CASE WHEN rank=1 THEN 'yes' ELSE 'no' END)` which gives `no` as result in case multiple rows. But writing as `MIN(CASE WHEN rank=1 THEN 'yes' END)` as gives 'yes` as result when condition is satisfied. This is  just applying min on "one `yes` and multiple `NULLs`"

Reply:
Here I want to convert long format to wide format.

from 
```
awc_id, cbe_date, theme_cbe, 
awc_id1, 2019-05-03, xyz,  
awc_id1, 2019-05-12, pqr,

```

TO

```
awc_id, cbe_date_1, theme_cbe1, cbe_date_2, theme_cbe2
awc_id1, 2019-05-03, xyz, 2019-05-12, pqr

```

I could have done this through two separate queries. but one query looks better than two similar queries. Feel free to disagree if you think this can be done in other way than what is currently implemented.

Reply:
> This is just applying min on "one yes and multiple NULLs"

Does this imply that `MIN('yes', NULL) == MAX('yes', NULL) == 'yes'`?

I find that weird behavior but I believe you

Reply:
answer seems to be yes.
```
postgres=# CREATE TABLE cz_agg_test ( col text);
CREATE TABLE
postgres=# insert into cz_agg_test VALUES ('yes');
INSERT 0 1
postgres=# insert into cz_agg_test VALUES (NULL);
INSERT 0 1
postgres=# select * from cz_agg_test;
 col 
-----
 yes
 
(2 rows)

postgres=# select MIN(col) from cz_agg_test;
 min 
-----
 yes
(1 row)

postgres=# select MAX(col) from cz_agg_test;
 max 
-----
 yes
(1 row)

```

Reply:
yeah. Sure I can just remove it